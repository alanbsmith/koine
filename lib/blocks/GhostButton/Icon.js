import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { Icon as BasicIcon } from '../../elements';

const modifierConfig = {
  left: (props) => `
    padding-right: ${rem(props.theme.dimensions.basePad)};
  `,
  right: (props) => `
    padding-left: ${rem(props.theme.dimensions.basePad)};
  `,
};

const Icon = styled(BasicIcon)`
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
