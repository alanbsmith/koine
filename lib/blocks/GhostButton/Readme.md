```jsx
<FlexContainer>
  <FlexRow>
    <FlexColumn>
      <GhostButton modifiers={['hoverDanger']}>
        <GhostButton.Icon name="home" modifiers={['left']} />
        <GhostButton.Text>action</GhostButton.Text>
      </GhostButton>
    </FlexColumn>
    <FlexColumn>
      <GhostButton>
        <GhostButton.Text>action</GhostButton.Text>
        <GhostButton.Icon name="home" modifiers={['right']} />
      </GhostButton>
    </FlexColumn>
  </FlexRow>
</FlexContainer>
```
