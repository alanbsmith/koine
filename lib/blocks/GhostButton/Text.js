import styled from 'styled-components';
import gridScale from '../../utils/gridScale';

const Text = styled.span`
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
