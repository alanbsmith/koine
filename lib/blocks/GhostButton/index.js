import styled from "styled-components";
import { applyStyleModifiers } from "styled-components-modifiers";

import Button from "../Button";
import Icon from "./Icon";
import Text from "./Text";

const modifierConfig = {
  hoverCustom: ({ theme, hoverColor }) => `
    &:hover, &:focus {
      border-color: ${theme.colors.base[hoverColor]};
      color: ${theme.colors.base[hoverColor]};
    }
  `,

  hoverDanger: ({ theme }) => `
    &:hover, &:focus {
      border-color: ${theme.colors.status.danger};
      color: ${theme.colors.status.danger};
    }
  `,

  hoverInfo: ({ theme }) => `
    &:hover, &:focus {
      border-color: ${theme.colors.status.info};
      color: ${theme.colors.status.info};
    }
  `,

  hoverSuccess: ({ theme }) => `
    &:hover, &:focus {
      border-color: ${theme.colors.status.success};
      color: ${theme.colors.status.success};
    }
  `,

  hoverWarning: ({ theme }) => `
    &:hover, &:focus {
      border-color: ${theme.colors.status.warning};
      color: ${theme.colors.status.warning};
    }
  `,
  fullWidth: () => `flex: 1`
};

const GhostButton = styled(Button)`
  background: transparent;
  border-color: ${props => props.theme.colors.base.chrome600};
  color: ${props => props.theme.colors.base.chrome600};

  &:hover,
  &:focus {
    background: transparent;
    border-color: ${props => props.theme.colors.base.chrome900};
    box-shadow: none;
    color: ${props => props.theme.colors.base.chrome900};
  }

  ${applyStyleModifiers(modifierConfig)};
`;

GhostButton.defaultProps = {
  hoverColor: "chrome900"
};

GhostButton.Icon = Icon;
GhostButton.Text = Text;

/** @component */
export default GhostButton;
