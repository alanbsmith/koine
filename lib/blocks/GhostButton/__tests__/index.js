import React from 'react';
import GhostButton from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('GhostButton Block', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <GhostButton>
        <GhostButton.Text>Login</GhostButton.Text>
        <GhostButton.Icon name="home" />
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the bgDanger modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['bgDanger']}>
        <GhostButton.Text>Login</GhostButton.Text>
        <GhostButton.Icon name="home" />
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverBgDanger modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['hoverBgDanger']}>
        <GhostButton.Text>Login</GhostButton.Text>
        <GhostButton.Icon name="home" />
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the fullWidth modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['fullWidth']}>
        <GhostButton.Text>Login</GhostButton.Text>
        <GhostButton.Icon name="home" />
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverCustom modifier', () => {
    const tree = renderWithTheme(
      <GhostButton
        hoverColor="pink"
        modifiers={['hoverCustom']}
      >
        <GhostButton.Text>Login</GhostButton.Text>
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses default styles if no hoverColor is provided with hoverCustom modifier', () => {
    const tree = renderWithTheme(
      <GhostButton
        modifiers={['hoverCustom']}
      >
        <GhostButton.Text>Login</GhostButton.Text>
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('border-color', '#8492A6');
  });

  it('adds the hoverDanger modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['hoverDanger']}>
        <GhostButton.Text>Login</GhostButton.Text>
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverInfo modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['hoverInfo']}>
        <GhostButton.Text>Login</GhostButton.Text>
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverSuccess modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['hoverSuccess']}>
        <GhostButton.Text>Login</GhostButton.Text>
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverWarning modifier', () => {
    const tree = renderWithTheme(
      <GhostButton modifiers={['hoverWarning']}>
        <GhostButton.Text>Login</GhostButton.Text>
      </GhostButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
