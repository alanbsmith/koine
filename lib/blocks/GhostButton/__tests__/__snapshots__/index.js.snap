// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GhostButton Block adds the bgDanger modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #FF4949;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c3 {
  font-size: 1rem;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c3"
  />
</button>
`;

exports[`GhostButton Block adds the fullWidth modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c3 {
  font-size: 1rem;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c3"
  />
</button>
`;

exports[`GhostButton Block adds the hoverBgDanger modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c1:hover,
.c1:focus {
  background-color: #ff6262;
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c3 {
  font-size: 1rem;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c3"
  />
</button>
`;

exports[`GhostButton Block adds the hoverCustom modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c0:hover,
.c0:focus {
  border-color: undefined;
  color: undefined;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
</button>
`;

exports[`GhostButton Block adds the hoverDanger modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c0:hover,
.c0:focus {
  border-color: #FF4949;
  color: #FF4949;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
</button>
`;

exports[`GhostButton Block adds the hoverInfo modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c0:hover,
.c0:focus {
  border-color: #1FB6FF;
  color: #1FB6FF;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
</button>
`;

exports[`GhostButton Block adds the hoverSuccess modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c0:hover,
.c0:focus {
  border-color: #13CE66;
  color: #13CE66;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
</button>
`;

exports[`GhostButton Block adds the hoverWarning modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c0:hover,
.c0:focus {
  border-color: #FFC82C;
  color: #FFC82C;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
</button>
`;

exports[`GhostButton Block renders correctly 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

.c3 {
  font-size: 1rem;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c3"
  />
</button>
`;

exports[`GhostButton Block uses default styles if no hoverColor is provided with hoverCustom modifier 1`] = `
.c0 {
  background: transparent;
  border-color: #646464;
  color: #646464;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-color: #191919;
  box-shadow: none;
  color: #191919;
}

.c0:hover,
.c0:focus {
  border-color: #191919;
  color: #191919;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c1:hover,
.c1:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0 c1"
>
  <span
    className="c2"
  >
    Login
  </span>
</button>
`;
